dali/test/python/test_operator_slice.py

```diff
@@ -418,3 +398,126 @@ def test_slice_vs_numpy():
+            sliced = [abs_slice_shape[i] - pad_before[i] - pad_after[i] for i in range(naxes)]
+
+            if out_of_bounds_policy == 'trim_to_shape':
+                assert(all([pad_before[i] == 0 for i in range(naxes)]))
+                assert(all([pad_after[i] == 0 for i in range(naxes)]))
+                assert(all([sliced[i] == out_shape[i] for i in range(naxes)]))
+
+            for i in range(out_shape[0]):
+                for j in range(out_shape[1]):
+                    if (i >= pad_before[0] and j >= pad_before[1] and i < pad_before[0] + sliced[0] and j < pad_before[1] + sliced[1]):
+                        assert((sample_out[i, j, :] == sample_in[abs_start[0] + i, abs_start[1] + j, :]).all())
+                    else:
+                        assert((sample_out[i, j, :] == fill_values).all())
+
+def test_slice_with_out_of_bounds_policy_support():
+    in_shape = (40, 80, 3)
+    for out_of_bounds_policy in ['pad', 'trim_to_shape']:
+        for device in ['gpu', 'cpu']:
+            for batch_size in [1, 3]:
+                for normalized_anchor, normalized_shape in [(False, False), (True, True)]:
+                    for fill_values in [None, (0x76, 0xb0, 0x00)]:
+                        yield check_slice_with_out_of_bounds_policy_support, \
+                            device, batch_size, in_shape, out_of_bounds_policy, fill_values, \
+                            normalized_anchor, normalized_shape
+
+def check_slice_with_out_of_bounds_error(device, batch_size, input_shape=(100, 200, 3),
+                                         normalized_anchor=False, normalized_shape=False):
+    # This test case is written with HWC layout in mind and "HW" axes in slice arguments
+    axis_names = "HW"
+    naxes = len(axis_names)
+    axes = None
+    layout = "HWC"
+    assert(len(input_shape) == 3)
+
+    eii = RandomDataIterator(batch_size, shape=input_shape)
+    eii_arg = SliceArgsIterator(batch_size, len(input_shape), image_shape=input_shape,
+                                image_layout=layout, axes=axes, axis_names=axis_names,
+                                normalized_anchor=normalized_anchor,
+                                normalized_shape=normalized_shape,
+                                min_norm_anchor=-0.5, max_norm_anchor=-0.1,
+                                min_norm_shape=1.1, max_norm_shape=3.6)
+    pipe = SliceSynthDataPipeline(device, batch_size, layout, iter(eii), iter(eii_arg),
+                                  axes=axes, axis_names=axis_names,
+                                  normalized_anchor=normalized_anchor,
+                                  normalized_shape=normalized_shape,
+                                  out_of_bounds_policy="error")
+
+    pipe.build()
+    with assert_raises(RuntimeError):
+        outs = pipe.run()
+
+def test_slice_with_out_of_bounds_error():
+    in_shape = (40, 80, 3)
```

Comment:
Could there be any reason to test slice with non-uniform batch?

Reply:
Doesn't make much sense for this particular test case. Do you mean generally speaking?